sct-commons/src/test/java/org/lfenergy/compas/sct/commons/scl/SclServiceTest.java

```diff
@@ -262,49 +263,129 @@ void testGetExtRefSourceInfo() throws Exception {
         extRefInfo.setHolderLDInst(ldInst);
         extRefInfo.setHolderLnClass(lnClass);
 
+        //When
         var controlBlocks = SclService.getExtRefSourceInfo(scd, extRefInfo);
-        assertEquals(2, controlBlocks.size());
-        controlBlocks.forEach(controlBlock -> assertTrue(
-                        controlBlock.getName().equals("goose1") || controlBlock.getName().equals("smv1")
-                )
-        );
+
+        //Then
+        assertEquals(0, controlBlocks.size());
+    }
+
+    @Test
+    void getExtRefSourceInfo_shouldReturnListOfControlBlocks_whenExtRefMatchFCDA() throws Exception {
+        //Given
+        SCL scd = SclTestMarshaller.getSCLFromFile("/scl-srv-scd-extref-cb/issue_175_scd_get_cbs_test.xml");
+        String iedName = "IED_NAME2";
+        String ldInst = "LD_INST21";
+        String lnClass = TLLN0Enum.LLN_0.value();
+        SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
+        IEDAdapter iedAdapter = sclRootAdapter.getIEDAdapterByName(iedName);
+        LDeviceAdapter lDeviceAdapter = assertDoesNotThrow(() -> iedAdapter.getLDeviceAdapterByLdInst(ldInst).get());
+        LN0Adapter ln0Adapter = lDeviceAdapter.getLN0Adapter();
+        List<TExtRef> extRefs = ln0Adapter.getExtRefs(null);
+        assertFalse(extRefs.isEmpty());
+
+        ExtRefInfo extRefInfo = new ExtRefInfo(extRefs.get(0));
+
+        extRefInfo.setHolderIEDName(iedName);
+        extRefInfo.setHolderLDInst(ldInst);
+        extRefInfo.setHolderLnClass(lnClass);
+
+        //When
+        var controlBlocks = SclService.getExtRefSourceInfo(scd, extRefInfo);
```

Comment:
Don't use `var` here because the method name is misleading. I expected `getExtRefSourceInfo` to return `ExtRefSourceInfo` but returns a list of`ControlBlock`.